File: admin/types/textbook.ts
declare global {
  interface Textbook {
    id: number;
    subject: string;
    version: string;
    grade: number;
    semester: string;
    file?: string;
    index_file_id?: string;
    is_parsed?: number;
    status: number;
    create_time: number;
    update_time?: number;
  }

  interface TextbookSearchParams extends SearchParams {
    keyword?: string;
    subject?: string;
    grade?: number;
  }

  interface TextbookForm {
    subject: string;
    version: string;
    grade: number;
    semester: string;
  }

  interface Unit {
    id: number;
    textbook_id: number;
    name: string;
    content: string;
    status: number;
    create_time: number;
    update_time?: number;
  }

  interface UnitSearchParams extends SearchParams {
    keyword?: string;
  }

  interface UnitForm {
    textbook_id: number;
    name: string;
    content: string;
  }

  interface UnitUpdateForm {
    name?: string;
    content?: string;
    status?: number;
  }

  interface Knowledge {
    id: number;
    textbook_id: number;
    unit_id: number;
    name: string;
    content: string;
    difficulty?: string;
    importance?: number;
    order?: number;
    status: number;
    create_time: number;
    update_time?: number;
  }

  interface KnowledgeSearchParams extends SearchParams {
    keyword?: string;
  }

  interface KnowledgeForm {
    textbook_id: number;
    unit_id: number;
    name: string;
    content: string;
    difficulty?: string;
    importance?: number;
    order?: number;
  }

  interface KnowledgeUpdateForm {
    name?: string;
    content?: string;
    difficulty?: string;
    importance?: number;
    order?: number;
    status?: number;
  }

  interface TextbookContentForm {
    name: string;
    content: string;
    unit_id?: number;
    textbook_id?: number;
  }
}

export {};

File: apps/server/shared/core/schema/textbook.py
"""
教材相关 Schema
"""

from typing import Optional

from pydantic import BaseModel


class TextbookSchema(BaseModel):
    """教材信息"""

    id: int
    subject: str
    version: str
    grade: int
    semester: str
    file: Optional[str] = None
    index_file_id: Optional[str] = None
    is_parsed: int = 0

    model_config = {"from_attributes": True}


class TeacherBookSchema(BaseModel):
    """教师用书信息"""

    id: int
    subject: str
    version: str
    grade: int
    semester: str
    file: Optional[str] = None
    index_file_id: Optional[str] = None

    model_config = {"from_attributes": True}


class UnitSchema(BaseModel):
    """单元信息"""

    id: int
    textbook_id: int
    name: str
    content: str
    textbook: Optional["TextbookSchema"] = None

    model_config = {"from_attributes": True}


__all__ = [
    "TextbookSchema",
    "TeacherBookSchema",
    "UnitSchema",
]
